src/domain/safe/safe.repository.ts

```diff
@@ -34,382 +34,382 @@ export class SafeRepository implements ISafeRepository {
+  async getTransactionQueueByModified(args: {
+    chainId: string;
+    safe: Safe;
+    limit?: number;
+    offset?: number;
+  }): Promise<Page<MultisigTransaction>> {
+    return this._getTransactionQueue({
+      chainId: args.chainId,
+      safe: args.safe,
+      ordering: '-modified',
+      limit: args.limit,
+      offset: args.offset,
+    });
   }
 
-  private async _getTransactionQueue(
-    chainId: string,
-    safe: Safe,
-    ordering: string,
-    limit?: number,
-    offset?: number,
-  ): Promise<Page<MultisigTransaction>> {
+  private async _getTransactionQueue(args: {
+    chainId: string;
+    safe: Safe;
+    ordering: string;
+    limit?: number;
+    offset?: number;
+  }): Promise<Page<MultisigTransaction>> {
     const transactionService =
-      await this.transactionApiManager.getTransactionApi(chainId);
+      await this.transactionApiManager.getTransactionApi(args.chainId);
     const page: Page<MultisigTransaction> =
-      await transactionService.getMultisigTransactions(
-        safe.address,
-        ordering,
-        false,
-        true,
-        undefined,
-        undefined,
-        undefined,
-        undefined,
-        undefined,
-        safe.nonce,
-        limit,
-        offset,
-      );
+      await transactionService.getMultisigTransactions({
+        safeAddress: args.safe.address,
+        ordering: args.ordering,
+        executed: false,
+        trusted: true,
+        nonceGte: args.safe.nonce,
+        limit: args.limit,
+        offset: args.offset,
+      });
     return this.multisigTransactionValidator.validatePage(page);
   }
 
-  async getCreationTransaction(
-    chainId: string,
-    safeAddress: string,
-  ): Promise<CreationTransaction> {
+  async getCreationTransaction(args: {
+    chainId: string;
+    safeAddress: string;
+  }): Promise<CreationTransaction> {
     const transactionService =
-      await this.transactionApiManager.getTransactionApi(chainId);
+      await this.transactionApiManager.getTransactionApi(args.chainId);
     const createTransaction = await transactionService.getCreationTransaction(
-      safeAddress,
+      args.safeAddress,
     );
     return this.creationTransactionValidator.validate(createTransaction);
   }
 
-  async getTransactionHistoryByExecutionDate(
-    chainId: string,
-    safeAddress: string,
-    limit?: number,
-    offset?: number,
-  ): Promise<Page<Transaction>> {
-    return this.getAllExecutedTransactions(
-      chainId,
-      safeAddress,
-      'executionDate',
-      limit,
-      offset,
-    );
+  async getTransactionHistoryByExecutionDate(args: {
+    chainId: string;
+    safeAddress: string;
+    limit?: number;
+    offset?: number;
+  }): Promise<Page<Transaction>> {
+    return this.getAllExecutedTransactions({
+      chainId: args.chainId,
+      safeAddress: args.safeAddress,
+      ordering: 'executionDate',
+      limit: args.limit,
+      offset: args.offset,
+    });
```

Comment:
We could also spread here.

Reply:
[9db8374](https://github.com/safe-global/safe-client-gateway-nest/pull/610/commits/9db8374f600be9d8fbe58d5c91761aef0ea8b5cf)